test/libsolidity/ViewPureChecker.cpp

```diff
@@ -109,6 +109,7 @@ BOOST_AUTO_TEST_CASE(environment_access)
 		"block.difficulty",
 		"block.number",
 		"block.gaslimit",
+		"gasleft()",
```

Comment:
@ekpyron @chriseth I guess this should have been `gasleft` without parentheses, but the code still compiles (`var x = gasleft; x;` vs `var x = gasleft(); x;`)

Reply:
`block.blockhash(7)` above also uses parentheses. We should have tests for both versions if such a magic global is a function.

Reply:
Oh cool, I think it must be correct this way, the others are variables, while these two are functions.